pirk-core/src/main/java/org/apache/pirk/test/distributed/testsuite/DistTestSuite.java

```diff
@@ -0,0 +1,334 @@
+    SystemConfiguration.setProperty("pirTest.embedSelector", "true");
+    BaseTests.testDNSHostnameQuery(dataElements, fs, MAPREDUCE, true, 1);
+    BaseTests.testSRCIPQuery(dataElements, fs, MAPREDUCE, true, 2);
+    BaseTests.testDNSIPQuery(dataElements, fs, MAPREDUCE, true, 1);
+
+    SystemConfiguration.setProperty("pirTest.embedSelector", "false");
+    BaseTests.testDNSHostnameQuery(dataElements, fs, MAPREDUCE, true, 2);
+    BaseTests.testDNSIPQuery(dataElements, fs, MAPREDUCE, true, 2);
+
+    // Change query for NXDOMAIN
+    SystemConfiguration.setProperty("pir.esQuery", "?q=rcode:3");
+
+    SystemConfiguration.setProperty("pirTest.embedSelector", "true");
+    BaseTests.testDNSNXDOMAINQuery(dataElements, fs, MAPREDUCE, true, 3);
+    SystemConfiguration.setProperty("pirTest.embedSelector", "false");
+    BaseTests.testDNSNXDOMAINQuery(dataElements, fs, MAPREDUCE, true, 3);
+
+    logger.info("Completed testESInputMR");
+  }
+
+  public static void testESInputSpark(FileSystem fs, List<JSONObject> dataElements) throws Exception
+  {
+    logger.info("Starting testESInputSpark");
+
+    SystemConfiguration.setProperty("pirTest.useHDFSExpLookupTable", "false");
+    SystemConfiguration.setProperty("pirTest.useExpLookupTable", "false");
+
+    SystemConfiguration.setProperty("pir.limitHitsPerSelector", "false");
+    SystemConfiguration.setProperty("pir.maxHitsPerSelector", "1000");
+
+    SystemConfiguration.setProperty("pir.allowAdHocQuerySchemas", "false");
+    SystemConfiguration.setProperty("pir.embedQuerySchema", "false");
+
+    // Set up ES configs
+    SystemConfiguration.setProperty("pir.dataInputFormat", InputFormatConst.ES);
+    SystemConfiguration.setProperty("pir.esQuery", "?q=rcode:0");
+    SystemConfiguration.setProperty("pir.esResource", SystemConfiguration.getProperty(DistributedTestDriver.ES_INPUT_RESOURCE_PROPERTY));
+
+    // Run tests
+    SystemConfiguration.setProperty("pirTest.embedSelector", "true");
+    BaseTests.testDNSHostnameQuery(dataElements, fs, SPARK, true, 1);
+    BaseTests.testDNSIPQuery(dataElements, fs, SPARK, true, 1);
+    BaseTests.testSRCIPQuery(dataElements, fs, SPARK, true, 2);
+
+    SystemConfiguration.setProperty("pirTest.embedSelector", "false");
+    BaseTests.testDNSHostnameQuery(dataElements, fs, SPARK, true, 2);
+    BaseTests.testDNSIPQuery(dataElements, fs, SPARK, true, 2);
+
+    // Change query for NXDOMAIN
+    SystemConfiguration.setProperty("pir.esQuery", "?q=rcode:3");
+
+    SystemConfiguration.setProperty("pirTest.embedSelector", "true");
+    BaseTests.testDNSNXDOMAINQuery(dataElements, fs, SPARK, true, 3);
+    SystemConfiguration.setProperty("pirTest.embedSelector", "false");
+    BaseTests.testDNSNXDOMAINQuery(dataElements, fs, SPARK, true, 3);
+
+    logger.info("Completed testESInputSpark");
+  }
+
+  public static void testSparkStreaming(FileSystem fs, List<JSONObject> pirDataElements) throws Exception
+  {
+    // testJSONInputSparkStreaming(fs, pirDataElements);
```

Comment:
That was some intermediate work, test* functions are now dead code.  I generalized testJSONInput(String platform, FileSystem fs, List<JSONObject> dataElements and testESInput(String platform, FileSystem fs, List<JSONObject> dataElements).

I didn't delete just yet as I was using for reference.  Will do in final PR.